src/Nethermind/Nethermind.Blockchain/Find/BlockExtensions.cs

```diff
@@ -125,6 +125,36 @@ public static class BlockExtensions
             }
         }
 
+        public static IEnumerable<(long LogIndex, long TransactionLogIndex, TxReceipt Receipt, LogEntry Entry)> FindLogsWithReceipts(
+            this BlockHeader blockHeader,
+            TxReceipt[] receipts,
+            LogFilter logFilter,
+            FindOrder receiptFindOrder = FindOrder.Ascending, // iterating forwards, by default we are interested in all items in order of appearance 
+            FindOrder logsFindOrder = FindOrder.Ascending) // iterating forwards, by default we are interested in all items in order of appearance
+        {
+            if (logFilter.Matches(blockHeader.Bloom))
+            {
+                int transactionLogIndex = 0;
+                for (int i = 0; i < receipts.Length; i++)
+                {
+                    TxReceipt receipt = GetItemAt(receipts, i, receiptFindOrder);
+                    if (logFilter.Matches(receipt.Bloom))
+                    {
+                        int logIndex = 0;
+                        for (int j = 0; j < receipt.Logs.Length; j++)
+                        {
+                            var receiptLog = GetItemAt(receipt.Logs, j, logsFindOrder);
+                            if (logFilter.Accepts(receiptLog))
+                            {
+                                yield return (logIndex++, transactionLogIndex, receipt, receiptLog);
+                            }
+                        }
+                        transactionLogIndex++;
```

Comment:
logIndexInTransaction

otherwise it means : index of transaction log
compoun nouns in english are read from back to front

Reply:
transactionLogIndex is a field in OE json RPC for logs, so lets treat it as a domain name.
On the other hand I think we got the indexes mixed up:

logIndex - log index in block
transactionLogIndex - log index in transaction
here its the other way around, also for log index in transaction you can just use indexer j

Reply:
Yeah, I misunderstood and used transactionLogIndex as transaction index in block and logIndex as log index in transaction... Now it's like you wrote - logIndex as log index in block and transactionLogIndex as log index in transaction. I can't use indexer j here, because inside the "for" loop there is one more "if" and not every iteration matches filter, and as we don't want to have gaps in indexes, I need to have one more incrementing